KerbalStuff/blueprints/mods.py

```diff
@@ -232,15 +229,23 @@ def edit_mod(mod_id: int, mod_name: str) -> Union[str, werkzeug.wrappers.Respons
         ckan = request.form.get('ckan')
         background = request.form.get('background')
         bgOffsetY = request.form.get('bg-offset-y', 0)
-        if not license or license == '':
-            return render_template("edit_mod.html", mod=mod, error="All mods must have a license.")
-        mod.short_description = short_description
         mod.license = license
         mod.donation_link = donation_link
         mod.external_link = external_link
         mod.source_link = source_link
-        mod.description = description
+        if isinstance(short_description, str):
+            mod.short_description = short_description.replace('\r\n', ' ').replace('\n', ' ')
+        else:
+            mod.short_description = ''
+        if isinstance(description, str):
+            mod.description = description.replace('\r\n', '\n')
+        else:
+            mod.description = ''
         mod.score = get_mod_score(mod)
+        if not license or license == '':
+            return render_template("edit_mod.html", mod=mod, error="All mods must have a license.")
+        if description == default_description:
```

Comment:
The result of the `.replace` goes into `mod.description`, but we're comparing `default_description` to `description`, which would still be the original unmodified string with `\r\n` in it.

Reply:
Also it looks like we are trying to apply validation checks _after_ we set properties on `mod`, committing the changes. We should validate before we commit.

Reply:
Hmm, I kinda did that on purpose to keep user changes as much as possible, so that users don't have to reselect the correct license if they don't edit the description, or, worse, retype the entire description if they left the license field empty.

I think I got an idea how we could solve that though.